README.md

```diff
@@ -189,9 +178,21 @@ Your database not listed here?
 
 <br>
 
+# How it works
+
+`data-diff` efficiently compares data using two modes:
+
+1. **`joindiff`**: Ideal for comparing data within the same database, utilizing outer joins for efficient row comparisons. It relies on the database engine for computation and has consistent performance.
```

Comment:
Can you convert these into toggles for both joindiff and hashdiff? Let's not force readers to context switch for a quick explainer. I agree with moving the how it works section here. 

## `joindiff`
- Recommended for comparing data within the same database
- Uses the outer join operation to diff the rows as efficiently as possible within the same database
- Fully relies on the underlying database engine for computation
- Requires both datasets to be queryable with a single SQL query
- Time complexity approximates JOIN operation and is largely independent of the number of differences in the dataset

## `hashdiff`
- Recommended for comparing datasets across different databases
- Can also be helpful in diffing very large tables with few expected differences within the same database
- Employs a divide-and-conquer algorithm based on hashing and binary search
- Can diff data across distinct database engines, e.g., PostgreSQL <> Snowflake
- Time complexity approximates COUNT(*) operation when there are few differences
- Performance degrades when datasets have a large number of differences

Reply:
Keep your new summary sentences though.

Reply:
That's a good option - did not think of toggles!

Reply:
re-added